src/index.d.ts

```diff
@@ -14,6 +14,30 @@ declare module 'bootstrap-vue' {
 
   export default VuePlugin
 
+  type VueVariant = 'active' | 'success' | 'info' | 'warning' | 'danger'
+  type SortDirection = 'asc' | 'desc'
+  type FormatterCallback = ((value: mixed, key: string, item: mixed) => string)
+
+  export interface TableField {
+    label?: string
+    headerTitle?: string
+    headerAbbr?: string
+    class?: string | string[]
+    formatter?: string | FormatterCallback
+    sortable: boolean
+    sortDirection: SortDirection
+    tdClass: string | string[] | FormatterCallback
+    thClass: string | string[]
+    thStyle: mixed
+    variant: VueVariant
+    tdAttr: mixed | ((value: mixed, key: string, item: mixed) => mixed)
+    isRowHeader: boolean
+  }
+  export interface TableFieldObject {
+    [key: string]: TableField
+  }
+  export interface TableFieldArray extends Array<string | {key: string & TableField}> {}
```

Comment:
`{key: string & TableField}` implies that `key` itself needs to be an object
Shouldn't this be:
`({key: string} & TableField)`

Reply:
Correct, thank you for that!